### PR TITLE
Initialize ANET database with command-line options

### PR DESCRIPTION
**Pull request by ce-simsoft**
Command line arguments are required for Initialization command instead of command prompt.

Closes [AB#666](https://dev.azure.com/ncia-anet/2aa083a5-af3d-44e1-8c7b-6e9e6b124d91/_workitems/edit/666)

#### User changes
- none

#### Super User changes
- none

#### Admin changes
- none

#### System admin changes
When setting up a fresh installation of ANET, the `anet init` command is no longer interactive but takes its settings with command-line options (run `anet init` without options to see how to use it).
- [ ] anet.yml or anet-dictionary.yml needs change
- [ ] db needs migration
- [ ] documentation has changed
- [ ] graphql schema has changed

### Checklist
  - [x] Described the user behavior in PR body
  - [x] Referenced/updated all related issues
  - [x] commits follow a `repo#issue: Title` title format and [these 7 rules](https://chris.beams.io/posts/git-commit/)
  - [x] commits have a [clean history](https://epage.github.io/dev/commits/), otherwise PR may be squash-merged
  - [ ] Added and/or updated unit tests
  - [ ] Added and/or updated e2e tests
  - [ ] Added and/or updated data migrations
  - [ ] Updated documentation
  - [x] Resolved all build errors and warnings
  - [ ] Opened debt issues for anything not resolved here

### CODE REVIEW

File: init.yml [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+#### This is the configuration file for the ANET db initialization.
+####
+#### For Production use, replace all areas with ${NAME} with
+#### the correct values for your installation.
+####
+#### For Development use, create a file called `localSettings.gradle`
+#### And set each environment variable there
+####  ie: `run.environment('ANET_SMTP_SERVER', 'my-mail-server.example.com`)
+#### Do not make any changes to this file unless they apply to all
+#### developers!
```
Comment (midmarch) [inline init.yml:13]: I don't think these comments are applicable to this log file.

I suggest to replace this with the reason why this log configuration file is different.

**Review by midmarch — DISMISSED** [review]
To have an additional log configuration is fine, but to a copy of the entire ANET application configuration is a not a good idea.

File: build.gradle [hunk 2]
```diff
@@ -280,14 +280,26 @@ task dbWait(dependsOn: 'compileJava', type: JavaExec) {
...
 	group = "database runtime"
 	description = "Runs the ANET database initialization command."
 	classpath = sourceSets.main.runtimeClasspath
 	environment(run.environment)
 	mainClass = mainClassName
-	args = ["init", anetConfig]
-	standardInput = System.in
+	def cmdline = ["init"]
+	if (project.hasProperty("args")) {
+		cmdline.addAll(project.args.split("\\s+"))
```
Comment (gjvoosten) [inline build.gradle:297]: This won't allow spaces in the argument values; something like:
```
               cmdline.addAll(project.args.split("(?<!\\\\)\\s+").collect { a -> a.replace("\\ ", " ")})
```
would do that.

**Review by gjvoosten — CHANGES_REQUESTED** [review]

File: src/main/java/mil/dds/anet/InitializationCommand.java [hunk 3]
```diff
@@ -56,35 +70,29 @@ protected void run(Environment environment, Namespace namespace, AnetConfigurati
...
     Scanner scanner = new Scanner(System.in);
 
-    // Set Classification String
-    System.out.print("Classification String >>");
+    // Set Classification String as default
     AdminSetting classifString = new AdminSetting();
     classifString.setKey(AdminSettingKeys.SECURITY_BANNER_CLASSIFICATION.name());
-    classifString.setValue(scanner.nextLine());
+    classifString.setValue("NATO_UNCLASSIFIED");
     engine.getAdminDao().saveSetting(classifString);
```
Comment (gjvoosten) [inline src/main/java/mil/dds/anet/InitializationCommand.java:77]: Extract this into a private method; it's almost the same code 6 times over.

File: build.gradle [hunk 4]
```diff
@@ -280,14 +280,29 @@ task dbWait(dependsOn: 'compileJava', type: JavaExec) {
 }
 dbWait.mustRunAfter('dockerStartDB')
 
+// Create the task that runs the initialization command. Run with:
+//   ./gradlew dbInit -Pargs="--adminOrgName <adminOrgName> --adminPosName <adminPosName>
+//                            --adminFullName <LASTNAME, Firstname> --adminDomainName <adminDomainName>"
+// e.g. (take care to escape spaces in argument values with a backslash):
+//   ./gradlew dbInit -Pargs="--adminOrgName ANET\ Administrators --adminPosName ANET\ Administrator
+//                            --adminFullName ADMIN,\ Arthur --adminDomainName arthur"
```
Comment (gjvoosten) [inline build.gradle:288]: In `insertBaseData-psql.sql` he's called *DMIN, Arthur*…

**Review by gjvoosten — CHANGES_REQUESTED** [review]
This PR seems to be about AB#666; please update the description and the issue link.`

**Review by gjvoosten — APPROVED** [review]

**Pull request merged**